core/state/database.go

```diff
@@ -64,6 +64,28 @@ type Database interface {
 
 	// TrieDB retrieves the low level trie database used for data storage.
 	TrieDB() *trie.Database
+
+	StartVerkleTransition(originalRoot, translatedRoot common.Hash, chainConfig *params.ChainConfig, cancunBlock *big.Int)
+
+	EndVerkleTransition()
+
+	InTransition() bool
+
+	Transitionned() bool
+
+	SetCurrentAccountHash(hash common.Hash)
+
+	GetCurrentAccountHash() common.Hash
+
+	SetCurrentSlotHash(hash common.Hash)
+
+	GetCurrentSlotHash() common.Hash
+
+	SetStorageProcessed(bool)
+
+	GetStorageProcessed() bool
```

Comment:
These one looks pretty weird, like quite leaky abstractions; but probably it's just to move forward. So SGTM.